crates/factor-outbound-networking/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+use std::{collections::HashMap, sync::Arc};
+
+use anyhow::Context;
+use futures_util::{
+    future::{BoxFuture, Shared},
+    FutureExt,
+};
+use spin_factor_variables::VariablesFactor;
+use spin_factor_wasi::WasiFactor;
+use spin_factors::{
+    Factor, FactorInstancePreparer, InstancePreparers, PrepareContext, Result, SpinFactors,
+};
+use spin_outbound_networking::{AllowedHostsConfig, ALLOWED_HOSTS_KEY};
+
+pub struct OutboundNetworkingFactor;
+
+impl Factor for OutboundNetworkingFactor {
+    type AppConfig = AppConfig;
+    type InstancePreparer = InstancePreparer;
+    type InstanceState = ();
+
+    fn configure_app<Factors: SpinFactors>(
+        &self,
+        app: &spin_factors::App,
+        _ctx: spin_factors::ConfigureAppContext<Factors>,
+    ) -> Result<Self::AppConfig> {
+        // Extract allowed_outbound_hosts for all components
+        let component_allowed_hosts = app
+            .components()
+            .map(|component| {
+                Ok((
+                    component.id().to_string(),
+                    component
+                        .get_metadata(ALLOWED_HOSTS_KEY)?
+                        .unwrap_or_default()
+                        .into_boxed_slice()
+                        .into(),
+                ))
+            })
+            .collect::<Result<_>>()?;
+        Ok(AppConfig {
+            component_allowed_hosts,
+        })
+    }
+}
+
+#[derive(Default)]
+pub struct AppConfig {
+    component_allowed_hosts: HashMap<String, Arc<[String]>>,
+}
+
+type AllowedHostsFuture = Shared<BoxFuture<'static, Arc<anyhow::Result<AllowedHostsConfig>>>>;
```

Comment:
nit: I think I'd prefer for there to be a type alias `SharedFuture<T> = Shared<BoxFuture<'static, Arc<T>>>` so that the output of the future is still seen where it's used. I personally find completely opaque type aliases to be harder to read because I always end up looking at the definition to find out what type will actually be produced. 

For example:

```rust
pub struct InstancePreparer {
    allowed_hosts_future: SharedFuture<anyhow::Result<AllowedHostsConfig>>,
}

```

Reply:
`SharedFuture` sounds the same as `Shared` (which is a `Future`). It could be...`SharedFutureResult`...I guess?

Reply:
Done.